src/main/java/com/kryptnostic/api/v1/storage/DefaultStorageClient.java

```diff
@@ -379,7 +382,8 @@ public String appendObject( ObjectMetadata objectMetadata, String body ) throws
         EncryptedSearchSharingKey encryptedSearchSharingKey = privKey.calculateSharingKey( bridgeKey );
         List<MetadataRequest> mreq = prepareMetadata( metadata, encryptedSearchSharingKey );
         uploadMetadata( mreq );
-        return objectApi.appendObject( objectMetadata.getId(), blockToAppend ).getData();
+        AppendObjectRequest req = new AppendObjectRequest( body.length(), blockToAppend );
+        return objectApi.appendObject( objectMetadata.getId(), req ).getData();
```

Comment:
There's no need to create a new AppendObjectRequest. 

Server side you can append the length of the body to the existing length of the metadata and update it.